libraries/DataFlash/DataFlash_File.cpp

```diff
@@ -1039,12 +999,11 @@ void DataFlash_File::flush(void)
 
 void DataFlash_File::_io_timer(void)
 {
-    uint16_t _tail;
-    if (_write_fd == -1 || !_initialised || _open_error) {
+    if (_write_fd == -1 || !_initialized || _open_error) {
         return;
     }
 
-    uint16_t nbytes = BUF_AVAILABLE(_writebuf);
+    uint32_t nbytes = _writebuf.available();
```

Comment:
Yep, it is required here because of the data type switch and isn't related with that other change. One thing that it may be good to do (even in the old code) its to explicit cast to 'size_t' when used with ::write below.  Because size_t is signed (and its difficult but it can be 16bit in some very limited hardware). What do you think?